tests/test_bukuDb.py

```diff
@@ -488,6 +489,125 @@ def test_compactdb(setup):
     assert bdb.get_rec_by_id(2) == (2, 'https://test.com:8080', 'test', ',es,est,tes,test,', 'a case for replace_tag test', 0)
     assert bdb.get_rec_by_id(3) is None
 
+
+@pytest.mark.parametrize(
+    'index, low, high, is_range, delay_commit, empty_database, input_retval',
+    product(
+        [-1000000, -1, 0, 1, 3, 4, 1000000],
+        [-1000000, -1, 0, 1, 3, 4, 1000000],
+        [-1000000, -1, 0, 1, 3, 4, 1000000],
+        [False, True],
+        [False, True],
+        [False, True],
+        ['y', 'n'],
+    )
+)
+def test_delete_rec(setup, index, low, high, is_range, delay_commit, empty_database, input_retval):
+    """test method."""
+    def teardown():
+        os.environ['XDG_DATA_HOME'] = TEST_TEMP_DIR_PATH
+
+    def unexpected_test_args():
+        """function to run when args not meet any scenario."""
+        assert 0, ('\n'.join([
+            'Unexpected test args:',
+            'index:{}'.format(index),
+            'low:{}'.format(low),
+            'high:{}'.format(high),
+            'is_range:{}'.format(is_range),
+            'delay_commit:{}'.format(delay_commit),
+            'empty_database:{}'.format(empty_database),
+            'input_retval:{}'.format(input_retval),
+        ]))
+    # setup
+    bdb = BukuDb()
+    db_len = 0
+    if not empty_database:
+        for bookmark in TEST_BOOKMARKS:
+            bdb.add_rec(*bookmark)
+        db_len = len(TEST_BOOKMARKS)
+
+    # TESTCASE index, low or high is less than 0
+    if index < 0 or low < 0 or high < 0:
+        with pytest.raises(AssertionError):
```

Comment:
As per the last discussion we are not raising an AssertionError here. Can you please modify the check accordingly?